### PR TITLE
DevWs for SHINA SYSTEM containing containing SiHas Multipurpose Sensor

### PR DESCRIPTION
**Pull request by shinasys**

### CODE REVIEW

File: devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy [hunk 1]
```diff
@@ -97,7 +103,7 @@ def parse(String description) {
         map.unit = "%"
         map.translatable = true
     }
-
+
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy]: This empty tab is not necessary here.

File: devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy [hunk 2]
```diff
@@ -257,9 +275,12 @@ def configure() {
     }
 
     if (isDSM300()) {
-        configCmds += zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, POWER_CONFIGURATION_BATTERY_VOLTAGE_ATTRIBUTE, DataType.UINT8, 30, 21600, 0x01/*100mv*1*/)
         configCmds += zigbee.configureReporting(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS, DataType.BITMAP16, 0, 0xffff, null)
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy:278]: Since You adding the support for CSM300 in this PR, so why did You change the code for another model -> DSM300 ? Is this power configuration not needed anymore here ?

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy (hunk 2 above)
Comment (shinasys) [reply devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy:278]: I already called it above, so deleted it.

File: devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy [hunk 3]
```diff
@@ -79,6 +83,8 @@ def parse(String description) {
                 map = translateZoneStatus(zs)
             } else if (descMap?.clusterInt == OCCUPANCY_SENSING_CLUSTER && descMap.attrInt == OCCUPANCY_SENSING_OCCUPANCY_ATTRIBUTE && descMap?.value) {
                 map = getMotionResult(descMap.value == "01" ? "active" : "inactive")
+            } else if (descMap?.clusterInt == ANALOG_INPUT_BASIC_CLUSTER && descMap.attrInt == ANALOG_INPUT_BASIC_PRESENT_VALUE_ATTRIBUTE && descMap?.value) {
+                map = getAnalogInputResult(Long.parseLong(descMap.value, 16))
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy]: What's the example input here ? (descMap.value)
Did You try `Integer.parseInt(descMap.value,16)` ?
That way maybe You wouldn't have to parse it to long, convert to float and round this value in `getAnalogInputResult`

Comment (shinasys) [reply devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy]: This value is the number of people. I committed again.

File: devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy [hunk 4]
```diff
@@ -191,6 +197,18 @@ private Map getContactResult(value) {
 	]
 }
 
+private Map getAnalogInputResult(value) {
+    Float f = Float.intBitsToFloat(value.intValue())
+    int pc = f.round(0)
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy:202]: Since You call this method with the integer
`map = getAnalogInputResult(Integer.parseInt(descMap.value,16))`

Those operations seem to be unnecessary.

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy (hunk 4 above)
Comment (shinasys) [reply devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy:202]: This Type is DataType.FLOAT4. So i think need to conversion.

 example) 5 people count
   * 0000A040 ->  Integer.parseInt(descMap.value,16) ->  1084227584  ->  Float.intBitsToFloat(value.intValue()) -> 5

_

**Review by KKlimczukS — APPROVED** [review]
